patches/api/0007-Adventure.patch

```diff
@@ -3539,6 +3558,45 @@ index 4dba721aefe4fc6699b3b4bfa7ecb0b19c2a2a1a..01dec2c877df58c9dc22445e8b1f9ce2
 +    public @NotNull MapCursor addCursor(int x, int y, byte direction, byte type, boolean visible, @Nullable net.kyori.adventure.text.Component caption) {
 +        return addCursor(new MapCursor((byte) x, (byte) y, direction, type, visible, caption));
 +    }
++    // Paper end
+ }
+diff --git a/src/main/java/org/bukkit/permissions/Permissible.java b/src/main/java/org/bukkit/permissions/Permissible.java
+index 228421154913116069c20323afb519bdde2134df..c0c58024a8213863184964d58eeffc3d06c0eb54 100644
+--- a/src/main/java/org/bukkit/permissions/Permissible.java
++++ b/src/main/java/org/bukkit/permissions/Permissible.java
+@@ -126,4 +126,34 @@ public interface Permissible extends ServerOperator {
+      */
+     @NotNull
+     public Set<PermissionAttachmentInfo> getEffectivePermissions();
++
++    // Paper start - add TriState permission checks
++    /**
++     * Checks if this object has a permission set and, if it is set, the value of the permission.
++     *
++     * @param permission the permission to check
++     * @return a tri-state of if the permission is set and, if it is set, it's value
++     */
++    default net.kyori.adventure.util.@NotNull TriState checkPermission(final @NotNull Permission permission) {
```

Comment:
I'd call this `permissionValue`, but it's not hugely important. what do others think?

Reply:
I went for check as the pointer is called a checker, but the method on the checker is `value`. imo either one is fine tbh

Reply:
I went for check as the pointer is called a checker, but the method on the checker is `value`. imo either one is fine tbh